dracut/30ignition/module-setup.sh

```diff
@@ -56,5 +56,9 @@ install() {
 
 installkernel() {
     # We definitely need this one in the initrd to support Ignition cfgs on qemu
-    instmods -c qemu_fw_cfg
+    if grep -q CONFIG_FW_CFG_SYSFS=m /usr/lib/modules/$kernel/config; then
+        instmods -c qemu_fw_cfg
+    elif ! grep -q CONFIG_FW_CFG_SYSFS=y /usr/lib/modules/$kernel/config; then
+        derror "Kernel compiled without CONFIG_FW_CFG_SYSFS, qemu provisioning will fail!"
```

Comment:
This only logs the message, it doesn't make it a hard error. I'm open to changing this to:

```
if ! grep CONFIG..=y; then
  instmods -c qemu_fw_cfg
fi
```

so that it just plain errors out if the kernel is built without it. But OTOH, we don't *absolutely* need that module either if the target kernel for some reason doesn't support it at all for whatever reason.

Reply:
I typically err on the side of hard error. I'd be interested to know why a target kernel wouldn't be able to support it. Are there common valid reasons?

Reply:
I was thinking about some targeted cases where one might use a slimmed down kernel e.g. IoT, but yeah totally fine just making it a req until the use case shows up.

Reply:
Updated! :arrow_down: